Classes/Domain/Repository/CommentRepository.php

```diff
@@ -246,12 +246,23 @@ protected function getFindByEmailAndPostIdConstraints(QueryInterface $query, $em
      *
      * @return object
      */
-    protected function getValidConstraints(QueryInterface $query)
+    protected function getValidConstraints(QueryInterface $query, $excludeHiddenPosts = false)
     {
         $constraints = $query->logicalAnd(
             $query->equals('spam', 0),
             $query->equals('approved', 1)
         );
+        // By default, comments of hidden posts will be included in the result, because there is no relation / join from comments to posts
```

Comment:
Correct. What is the use case for this (a list of all comments, including those of hidden posts, with populated posts)? The BE comment overview? I hadn't considered that one yet.

Couldn't this be fixed by temporarily changing the enablefields for the posts table? Could you give this a try?